### ISSUE TITLE
W&B stuck after epoch completed

### ISSUE THREAD

**Issue by khandriod** [ultralytics/yolov5#6652]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

I have custom trained the Yolov5s model. After the training finished for 256 epoch (I specified 300 epoch, don't know why it didn't run full epoch) the process stops at W&B  logging step. 
This is the error. It's been stuck there for hours. What Should I do? How to exit this step?
`wandb: Waiting for W&B process to finish, PID 2057... (success).
wandb: / 146.61MB of 146.75MB uploaded (0.00MB deduped)`

### Environment

YOLOv5s
Dockerfile
Ubuntu-20.xx

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@khandriod sorry to hear about that! @AyushExel what does the `(0.00MB deduped)` message mean?

**Comment by AyushExel** [comment]
@glenn-jocher it means that no data was de-duped. Wand does some optimizations to reduce the amount to data needed to be uploaded. 
@khandriod can you share your run link, and the command to reproduce? Wandb sync only after the training process is finished. It should not stop in between training epochs

**Comment by khandriod** [comment]
Thanks, @glenn-jocher, and @AyushExel.   I don't know what is run link is, but I used Yolov5s for 300 epochs and it couldn't reach there, exited 256, and stuck at W&B. 
Well, I had to cancel and run it again, and it was running smoothly for 100 epochs.

**Comment by AyushExel** [comment]
@khandriod you can find the run link by going to wandb.ai. there you'll see all your yolov5 runs

**Comment by khandriod** [comment]
here it is
https://wandb.ai/jabbar/YOLOv5/runs/1ej8j7zk?workspace=user-jabbar

**Comment by AyushExel** [comment]
@glenn-jocher @khandriod  This doesn't seem like its crashed? Is there some early stopping at play?
<img width="1218" alt="Screenshot 2022-02-16 at 4 39 54 PM" src="https://user-images.githubusercontent.com/15766192/154253098-b73e6439-8a2f-48f7-864f-99e1a5a847dc.png">

**Comment by khandriod** [comment]
@AyushExel I have no idea

**Comment by AyushExel** [comment]
@glenn-jocher can you see the above log so confirm if the execution was stopped by early stopper logic? I don't see any crash tracebacks

**Comment by glenn-jocher** [comment]
@AyushExel looks like this for me. It's pretty hard to decipher what is going on due to the ongoing W&B Logs bug. It feels like I've been saying the same thing for a year now with no fix in sight.

@khandriod despite a W&B bug that affects the logs which makes them near impossible to read, I don't see any actual error messages on your screen. There is an EarlyStopping note which means that the YOLOv5 early stopper stopped training after no improvement being observed. 

<img width="1607" alt="Screenshot 2022-02-16 at 15 06 06" src="https://user-images.githubusercontent.com/26833433/154281026-0c3f1f84-a616-4f43-8580-553eec638535.png">

**Comment by glenn-jocher** [comment]
@AyushExel please do anything you can to fix the Logs bug, it's a tremendous pain point. Talk to Lucas if you have to.

**Comment by AyushExel** [comment]
@glenn-jocher Yeah it is very inconvenient. Sorry about that. I've filed the tickets for this already in the past. Let me push this again.

**Comment by glenn-jocher** [comment]
@AyushExel thanks!

**Comment by glenn-jocher** [comment]
@AyushExel thanks!

**Comment by bit-scientist** [comment]
Hi, @AyushExel. I also have the same issue here, `wandb: 1.783 MB of 1.795 MB uploaded (0.000 MB deduped)`, I ran this command `python train.py --weights yolov5m6.pt --data data/data.yaml --hyp data/hyps/hyp.scratch-med.yaml --batch-size 16  --img-size 640 --device 0 --cache`. The training crushed after I used `--cache` in some other training and that caused the lack of memory. I can't even stop it, I tried just closing the screen terminal and opened again, but no change.
